dbadapter/db_adapter.go

```diff
@@ -64,10 +86,10 @@ type PatchOperation struct {
 
 func setDBClient(url, dbname string) (DBInterface, error) {
 	mClient, errConnect := mongoapi.NewMongoClient(url, dbname)
-	if mClient.Client != nil {
-		return mClient, nil
+	if errConnect != nil {
+		return nil, errConnect
 	}
-	return nil, errConnect
+	return &MongoDBClient{MongoClient: *mClient}, nil
```

Comment:
Should not this just be as follows?
```suggestion
	return &MongoDBClient{*mClient}, nil
```